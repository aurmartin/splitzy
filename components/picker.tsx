import { Picker as RNPicker } from "@react-native-picker/picker";
import { View } from "react-native";
import Label from "./label";
import { Colors } from "@/lib/constants";
interface PickerProps<T> {
  selectedValue: T;
  onValueChange: (value: T) => void;
  children: React.ReactNode;
  label?: string;
}

export default function Picker<T>(props: PickerProps<T>) {
  const { selectedValue, onValueChange, children, label } = props;

  return (
    <View>
      {label && <Label>{label}</Label>}
      <View
        style={{
          borderWidth: 1,
          borderColor: Colors.secondary,
          borderStyle: "solid",
          borderRadius: 4,
          backgroundColor: "white",
        }}
      >
        <RNPicker
          testID="picker"
          selectedValue={selectedValue}
          onValueChange={onValueChange}
          style={{
            padding: 0,
            marginTop: -5,
            marginBottom: -5,
          }}
        >
          {children}
        </RNPicker>
      </View>
    </View>
  );
}
