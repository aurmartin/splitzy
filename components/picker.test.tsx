import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react-native";
import Picker from "./picker";
import { Picker as RNPicker } from "@react-native-picker/picker";

describe("Picker", () => {
  it("renders correctly", () => {
    const mockOnValueChange = jest.fn();
    render(
      <Picker
        label="Test Label"
        selectedValue="option1"
        onValueChange={mockOnValueChange}
      >
        <RNPicker.Item label="Option 1" value="option1" />
        <RNPicker.Item label="Option 2" value="option2" />
      </Picker>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("displays the label correctly", () => {
    const mockOnValueChange = jest.fn();
    render(
      <Picker
        label="Test Label"
        selectedValue="option1"
        onValueChange={mockOnValueChange}
      >
        <RNPicker.Item label="Option 1" value="option1" />
      </Picker>,
    );

    screen.getByText("Test Label");
  });

  it("handles value changes correctly", async () => {
    const mockOnValueChange = jest.fn();
    render(
      <Picker
        label="Test Label"
        selectedValue="option1"
        onValueChange={mockOnValueChange}
      >
        <RNPicker.Item label="Option 1" value="option1" />
        <RNPicker.Item label="Option 2" value="option2" />
      </Picker>,
    );

    fireEvent(screen.getByTestId("picker"), "onValueChange", "option2");
    expect(mockOnValueChange).toHaveBeenCalledWith("option2");
  });
});
