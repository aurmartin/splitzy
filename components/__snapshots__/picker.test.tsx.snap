// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker renders correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "marginBottom": 4,
        },
      ]
    }
  >
    Test Label
  </Text>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderColor": "hsl(0 0% 70%)",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": -5,
          "marginTop": -5,
          "padding": 0,
        }
      }
    >
      <RNCPicker
        accessibilityLabel="Test Label"
        items={
          [
            {
              "label": "Option 1",
              "testID": undefined,
              "textColor": undefined,
              "value": "option1",
            },
            {
              "label": "Option 2",
              "testID": undefined,
              "textColor": undefined,
              "value": "option2",
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="picker"
      />
    </View>
  </View>
</View>
`;
